cocotb/handle.py

```diff
@@ -651,14 +651,30 @@ def _set_value(self, value, call_sim):
                 The value to drive onto the simulator object.
 
         Raises:
-            TypeError: If target is not wide enough or has an unsupported type
-                 for value assignment.
+            TypeError: If target has an unsupported type for value assignment.
+
+            OverflowError: If int value is out of the range that can be represented
+                by the target. -2**(Nbits - 1) <= value <= 2**Nbits - 1
```

Comment:
No, there isn’t, at least not when working with simple logic vectors, rather than signed/unsigned types in vhdl or using signed in verilog.
I think we should allow setting the widest int range possible given the number of bits. If the user needs finer grained control, BinaryValue should be used instead of int.

Reply:
So vpi cannot detect the "signed" keyword on a signal declaration?

Reply:
I would not be surprised if vpi can detect this, but the point is that the RTL may be using ```std_logic_vector(7 downto 0)``` rather than ```signed(7 downto 0)``` or ```unsigned(7 downto 0)```, so we can’t infer if the vector is assuming signed or unsigned values.

Reply:
Just added some documentation to clarify the behaviour of assigning signals using ints to the user.